src/UnitConversions.hs

```diff
@@ -54,3 +60,42 @@ convertIngredientToImperial ingr =
     Quart   -> ingr
     Gallon  -> ingr
     Other _ -> ingr
+
+
+convertTemperatureToMetric = convertTemperature C
+convertTemperatureToImperial = convertTemperature F
+
+convertTemperature :: TempUnit -> String -> String
+convertTemperature u s = unpack $ foldl replaceTemperature (pack s) (fmap packText $ fmap convertReplacement (findReplacements s))
+  where packText (s1, s2) = (pack s1, pack s2)
+        replaceTemperature text (old, new) = replace old new text
+        convertReplacement = fmap $ show . toTempUnit u
+
+findReplacements :: String -> [(String, Temperature)]
+findReplacements = map parseRegexResult . findTemperatures
+  where parseRegexResult l = (l!!0, Temperature (read (l!!1)) (parseTempUnit (l!!2)))
+
+findTemperatures :: String -> [[String]]
+findTemperatures s = s =~  "(-?[0-9]{1,3}) ?°?(C|F)"
```

Comment:
One more suggestion: This should probably require some whitespace before and after, in order to not match a prefix of, say, `30 Fritos` or `850 Crabs` :crab:.

The current implementation would also match, e.g. `Not300FaTemperature`.

Seems my suggestion to not require a degree sign opens the door to more false positives. My bad for not thinking of that earlier!

Reply:
Pattern now excludes `850 Crabs`, but still includes stuff like `850 C!`. Because of the number of test cases, I changed the temperature tests a bit to make the individual test cases easier to see.
```
            testCases = [ ("250°C", [Temperature 250 C])
                        , ("10 °C", [Temperature 10 C])
                        , ("-10 °C", [Temperature (-10) C])
                        , ("abc 15 C def", [Temperature 15 C])
                        , ("5 C", [Temperature 5 C])
                        , ("250C", [Temperature 250 C])
                        , ("999°F", [Temperature 999 F])
                        , ("34F", [Temperature 34 F])
                        , ("250°C.", [Temperature 250 C])
                        , ("250°C!", [Temperature 250 C])
                        , ("250°C?", [Temperature 250 C])
                        , ("10C 15F", [Temperature 10 C, Temperature 15 F])
                        , ("0 Frites", [])
                        , ("250Crabs", [])
                        , ("250 Crabs", [])] :: [(String, [Temperature])] in
            sequence_ $ map (uncurry (@=?) . fmap getTemperatures . swap) testCases
```
Downside: Only one fail will be shown at any time, even if there are more.

Maybe you have some input? It feels like I am overseeing a built-in method for that.

Reply:
done